hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/TopologySubcommand.java

```diff
@@ -180,16 +204,119 @@ private String getAdditionNodeOutput(HddsProtos.Node node) {
 
   // Format "ipAddress(hostName):PortName1=PortValue1    OperationalState
   //     networkLocation
-  private void printNodesWithLocation(Collection<HddsProtos.Node> nodes) {
-    nodes.forEach(node -> {
-      System.out.print(" " + getAdditionNodeOutput(node) +
-          node.getNodeID().getIpAddress() + "(" +
-          node.getNodeID().getHostName() + ")" +
-          ":" + formatPortOutput(node.getNodeID().getPortsList()));
-      System.out.println("    "
-          + node.getNodeOperationalStates(0) + "    " +
-          (node.getNodeID().getNetworkLocation() != null ?
-              node.getNodeID().getNetworkLocation() : "NA"));
-    });
+  private void printNodesWithLocation(Collection<HddsProtos.Node> nodes,
+                                      String state) throws IOException {
+    if (json) {
```

Comment:
Please merge multiple if else.

if (json && fullInfo) {.... return;}
else if(json && !fullInfo {... return;}

Reply:
Thank you for the suggestion, I have incorporated it in my next commit.